src/unit_tests/logcollector/test_read_win_event_channel.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "shared.h"
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+
+#include <stdint.h>
+#include <sec_api/stdlib_s.h>
+#include <winerror.h>
+#include <winevt.h>
+
+
+char *get_message(EVT_HANDLE evt, LPCWSTR provider_name, DWORD flags);
+
+/* Setup & Teardown */
+
+static int group_setup(void ** state) {
+    test_mode = 1;
+    return 0;
+}
+
+static int group_teardown(void ** state) {
+    test_mode = 0;
+    return 0;
+}
+
+/* Wraps */
+
+/* Tests */
+
+void test_get_message_get_publisher_fail(void ** state) {
+    EVT_HANDLE evt = NULL;
+    LPCWSTR provider_name = L"provider_name";
+
+    expect_value(wrap_EvtOpenPublisherMetadata, Session, NULL);
+    expect_string(wrap_EvtOpenPublisherMetadata, PublisherId, provider_name);
+    expect_value(wrap_EvtOpenPublisherMetadata, LogFilePath, NULL);
+    expect_value(wrap_EvtOpenPublisherMetadata, Locale, 0);
+    expect_value(wrap_EvtOpenPublisherMetadata, Flags, 0);
+    will_return(wrap_EvtOpenPublisherMetadata, NULL);
+
+    will_return(wrap_GetLastError, ERROR_FILE_NOT_FOUND);
+    will_return(wrap_FormatMessage, "File not found.");
+    expect_string(__wrap__mdebug1, formatted_msg, "Could not EvtOpenPublisherMetadata() with flags (1) which returned (2): File not found.");
+
+    assert_null(get_message(evt, provider_name, EvtFormatMessageEvent));
+}
+
+void test_get_message_get_size_fail(void ** state) {
+    EVT_HANDLE evt = NULL;
+    LPCWSTR provider_name = L"provider_name";
+    char message[] = "Test_Message";
```

Comment:
Some variables and values of each test are repeated, can this be unified into a common setup stage?